Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,433 @@
+            throw DeployerPluginError.error(error)
+        } catch {
+            print("Unexpected error : \(error)")
+            throw DeployerPluginError.error(error)
+        }
+        
+    }
+    
+    private func findExecutable(context: PluginContext,
+                                executableName: String,
+                                helpMessage: String,
+                                verboseLogging: Bool) throws -> Path {
+        
+        guard let executable = try? context.tool(named: executableName) else {
+            print("Can not find `\(executableName)` executable.")
+            print(helpMessage)
+            throw DeployerPluginError.toolNotFound(executableName)
+        }
+        
+        if verboseLogging {
+            print("-------------------------------------------------------------------------")
+            print("\(executableName) executable : \(executable.path)")
+            print("-------------------------------------------------------------------------")
+        }
+        return executable.path
+    }
+    
+    private func validate(samExecutablePath: Path,
+                          samDeploymentDescriptorFilePath: String,
+                          verboseLogging: Bool) throws {
+        
+        print("-------------------------------------------------------------------------")
+        print("Validating SAM deployment descriptor")
+        print("-------------------------------------------------------------------------")
+        
+        do {
+            try Utils.execute(
+                executable: samExecutablePath,
+                arguments: ["validate",
+                            "-t", samDeploymentDescriptorFilePath,
+                            "--lint"],
+                logLevel: verboseLogging ? .debug : .silent)
+            
+        } catch let error as DeployerPluginError {
+            print("Error while validating the SAM template.")
+            print(error)
+            print("Run the deploy plugin again with --verbose argument to receive more details.")
+            throw DeployerPluginError.error(error)
+        } catch {
+            print("Unexpected error : \(error)")
+            throw DeployerPluginError.error(error)
+        }
+    }
+    
+    private func deploy(samExecutablePath: Path,
+                        samDeploymentDescriptorFilePath: String,
+                        stackName: String,
+                        verboseLogging: Bool) throws {
+        
+        //TODO: check if there is a samconfig.toml file.
+        // when there is no file, generate one with default data or data collected from params
+        
+        
+        print("-------------------------------------------------------------------------")
+        print("Deploying AWS Lambda function")
+        print("-------------------------------------------------------------------------")
+        do {
+            
+            try Utils.execute(
```

Comment:
does the user need to control any of these knobs?

Reply:
I can add a command line option to use a TOLML file instead.
The default behaviour of SAM CLI is to guide you through an interactive CLI at first run and save user's parameters in a TOML file.
The plugin could check for a TOML file (or one passed as parameter on the plugin command line) and skip these knobs when a TOML file exists.

What do you think ?

Reply:
sgtm

Reply:
Implemented.  New behaviour : 
- plugin generates samconfig.toml at first execution with sensible default values 
- plugin does not overwrite this file at later run. If developer modified it, it is kept and reused "as-is", unless --force option is passed on the plugin command line 
- plugin invoke sam command with minimum command line options, the rest is taken from the sam configuration file

Marking this as resolved, happy to dive deeper if you think it is necessary